packages/grid/x-data-grid-pro/src/hooks/features/columnPinning/useGridColumnPinning.tsx

```diff
@@ -71,69 +70,6 @@ export const useGridColumnPinning = (
 ): void => {
   const pinnedColumns = useGridSelector(apiRef, gridPinnedColumnsSelector);
   const theme = useTheme();
-  // Each visible row (not to be confused with a filter result) is composed of a central .MuiDataGrid-row element
-  // and up to two additional .MuiDataGrid-row's, one for the columns pinned to the left and another
-  // for those on the right side. When hovering any of these elements, the :hover styles are applied only to
-  // the row element that was actually hovered, not its additional siblings. To make it look like a contiguous row,
-  // this method adds/removes the .Mui-hovered class to all of the row elements inside one visible row.
-  const updateHoveredClassOnSiblingRows = React.useCallback(
-    (event: React.MouseEvent<HTMLElement>) => {
-      if (props.disableColumnPinning) {
-        return;
-      }
-
-      if (!Array.isArray(pinnedColumns.left) && !Array.isArray(pinnedColumns.right)) {
-        return;
-      }
-
-      const nbLeftPinnedColumns = pinnedColumns.left?.length ?? 0;
-      const nbRightPinnedColumns = pinnedColumns.right?.length ?? 0;
-      if (nbLeftPinnedColumns + nbRightPinnedColumns === 0) {
-        return;
-      }
-
-      const rowContainer = apiRef.current.virtualScrollerRef!.current!;
-      if (!rowContainer) {
-        return;
-      }
-
-      const index = event.currentTarget.dataset.rowindex;
-      const rowElements = rowContainer.querySelectorAll(
-        `.${gridClasses.row}[data-rowindex="${index}"]`,
-      );
-      rowElements.forEach((row) => {
-        // Ignore rows from other grid inside the hovered row
-        if (
-          row.closest(`.${gridClasses.virtualScroller}`) ===
-          apiRef.current.virtualScrollerRef!.current!
-        ) {
-          if (event.type === 'mouseenter') {
-            row.classList.add('Mui-hovered');
```

Comment:
Because we use direct DOM manipulations, after selecting the row it is rerendered and the hover class is gone.

Reply:
This approach has an advantage though - the code is colocated with the column-pinning feature code